### ISSUE TITLE
Support OTP login flow for guest users without AAD/MSA account (i.e. add ability to log-in to specific tenant)

### ISSUE THREAD

**Issue by edfreeman** [microsoft/AzureStorageExplorer#3561]
Currently, when adding an Azure account to the Storage Explorer, there is no "simple" way of authenticating as a guest user using a non AAD/MSA account using the [OTP authentication method](https://docs.microsoft.com/en-us/azure/active-directory/external-identities/one-time-passcode) (albeit, currently in preview). This functionality would be useful for scenarios when you're sharing data with external users who are authenticating via this method.

With the default authentication method in ASE, you get this error as soon as you enter a non-AAD/MSA account:

![image](https://user-images.githubusercontent.com/30260217/92727723-79457e80-f367-11ea-84ef-c75fc9ed6b6b.png)

Alternatively, enabling the (preview) MSAL authentication setting in the ASE settings, which opens a browser when authenticating initially leads to the same issue:

![image](https://user-images.githubusercontent.com/30260217/92728278-5c5d7b00-f368-11ea-8869-e9d684654b0a.png)

However, since at this point we have access to the auth url, I could replace the `organizations` part of the path with the `tenantId` for the target tenant. After doing this, I could enter my non AAD/MSA account email address and start the OTP process, and sign-in successfully:

![image](https://user-images.githubusercontent.com/30260217/92728884-692e9e80-f369-11ea-8ec3-0fd2633323ac.png)

**Describe the solution you'd like**

Ideally, when adding an Azure account, there'd be a way to specify the tenant you want to log-in to before starting the log-in process. This would construct the correct (specific) auth endpoint (which I've done manually above) and, in doing so, would support the OTP auth flow.

**Comment by davyhoyle** [comment]
Are there any plans to change the default authentication library to MSAL from the old ADAL - at present, we need to tell our users to enable this manually which most users don't find without help.

**Comment by jinglouMSFT** [comment]
Yes, once MSAL supports multi-MSA login we will then make using MSAL default. we are actively working with our partner team but there is no timeline yet.

**Comment by edfreeman** [comment]
@jinglouMSFT Will this include the ability to specify the tenant one wants to log-in to? (Which would presumably then enable the functionality outlined in this GitHub issue.)

**Comment by MRayermannMSFT** [comment]
@edfreeman @davyhoyle two things that may be of interest to y'all. In April 2021 we switched to MSAL by default. We also made it so you only log into your home tenant by default, and logging into other tenants is done explicitly by the user:
![image](https://user-images.githubusercontent.com/21112697/151067525-291f2715-008e-4cec-8476-3e2e850766f3.png)

How much do these changes help y'all? What additional things need to be done to move this issue along?

**Comment by edfreeman** [comment]
Hi @MRayermannMSFT ,

Unfortunately, the updates don't help with the use-case I outlined in the issue.

The problem is that non-AAD / MSA users don't have a "home" tenant, and are therefore not recognised on the initial login screen. These are "External Identities" which authenticate via the [Email one-time passcode](https://docs.microsoft.com/en-us/azure/active-directory/external-identities/one-time-passcode) flow. In my case, I'm trying to use a Gmail account to be able to access an ADLS filesystem that's been shared with me.

The main problem is that the initial log-in page when you try to authenticate to a "subscription" in ASE is hitting the <code>https://login.microsoftonline.com/<b>organizations</b>/oauth2/v2.0/authorize?...</code> endpoint. I believe this is the "common" AAD auth endpoint. This doesn't allow non-AAD / MSA emails to proceed through the sign-in journey.

If I change the URL manually to <code>https://login.microsoftonline.com/<b><target_tenant_id></b>/oauth2/v2.0/authorize?...</code> then this alters the AAD sign-in flow to point to a specific tenant, at which point I'm able to authenticate as my Gmail guest user.

So, the fix I see for this would be to allow the option to enter a target Tenant ID _**before**_ redirecting to the browser for authentication. Behind the scenes, this would  construct the second auth url I shared above, and would enable non-AAD/MSA accounts to log-in without having to hack around with the auth URL.

Of course, this assumes that the end-user knows the Tenant ID of the tenant they're trying to log-in to.

Hope that makes sense!
